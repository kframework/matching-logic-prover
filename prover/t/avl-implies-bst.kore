imports "include/prelude.kore"
imports "t/definitions/avl.kore"
imports "t/definitions/bst.kore"

claim \implies( \and( avl( H { ArrayIntInt }
                         , X { Int }
                         , F { SetInt }
                         , MIN { Int }
                         , MAX { Int }
                         , Height { Int }
                         , Balance { Int }
                         )
                    )
              , \and( bst( H { ArrayIntInt }
                         , X { Int }
                         , F { SetInt }
                         , MIN { Int }
                         , MAX { Int }
                         )
                    )
              )
strategy search-fol(bound: 3)
//strategy simplify . kt . simplify
//       . ( (right-unfold-Nth(0,0) . simplify . instantiate-existentials . smt-cvc4 )
//         | (right-unfold-Nth(0,1) . simplify . instantiate-existentials . smt-cvc4 )
//         | (instantiate-existentials . smt-cvc4 )
//         | (right-unfold-Nth(0,2) . simplify . instantiate-existentials . smt-cvc4 )
//         | noop
//         )
