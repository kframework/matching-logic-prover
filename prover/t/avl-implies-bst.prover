claim \implies( \and( avl( variable("H") { ArrayIntInt }
                         , variable("X") { Int }
                         , variable("F") { SetInt }
                         , variable("MIN") { Int }
                         , variable("MAX") { Int }
                         , variable("Height") { Int }
                         , variable("Balance") { Int }
                         , .Patterns
                         )
                    , .Patterns
                    )
              , \and( bst( variable("H") { ArrayIntInt }
                         , variable("X") { Int }
                         , variable("F") { SetInt }
                         , variable("MIN") { Int }
                         , variable("MAX") { Int }
                         , .Patterns
                         )
                    , .Patterns
                    )
              )
strategy search-bound(3)

// strategy simplify ; kt ; simplify
//        ; ( (right-unfold-Nth(0,0) ; simplify ; instantiate-existentials ; smt-cvc4 )
//          | (right-unfold-Nth(0,1) ; simplify ; instantiate-existentials ; smt-cvc4 )
//          | ( instantiate-existentials ; smt-cvc4 )
//          )
