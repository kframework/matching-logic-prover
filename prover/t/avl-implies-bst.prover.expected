<prover>
  <goal>
    <id>
      root
    </id>
    <active>
      true
    </active>
    <parent>
      .
    </parent>
    <k>
      \implies ( \and ( avl ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "F" ) { SetInt } , variable ( "MIN" ) { Int } , variable ( "MAX" ) { Int } , variable ( "Height" ) { Int } , variable ( "Balance" ) { Int } , .Patterns ) , .Patterns ) , \and ( bst ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "F" ) { SetInt } , variable ( "MIN" ) { Int } , variable ( "MAX" ) { Int } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
